### ISSUE TITLE
Update keyboard interaction for Select

### ISSUE THREAD

**Issue by jessiehuff** [patternfly/patternfly-react#3206]
Update the select component based on the keyboard interaction [epic](https://github.com/patternfly/patternfly-react/issues/3108), use the [design documentation](https://docs.google.com/document/d/1m9nICFRgCXnXeHYN487BMX9MHw4DjvOln2B5p8yJa70/edit#heading=h.bg94bjykerss) to ensure component meets requirements.

One of the most notable keyboard interaction changes that should be made is the checkboxes in Select. This is the only component that we use Enter to check and uncheck. We should update these to use Space like our other components and [expected keyboard interactions](https://webaim.org/techniques/keyboard/) for checkboxes. 

Write integration tests to prevent keyboard interaction regressions.

**Comment by nicolethoen** [comment]
This can be considered a duplicate of https://github.com/patternfly/patternfly-react/issues/6014 since all dropdown menus, including the select menu, will use the same Menu component under the hood, they will be using the same interaction patterns.
